### PR TITLE
Added homepage with apis and functionality

### PR DESCRIPTION
**Pull request by VLeads**
Added Homepage layout

Added video card for homepage

Integrated APIs for category and video card

Demo: https://rapid-fedvayg2v-vleads.vercel.app/

### CODE REVIEW

File: src/context/toast-context.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+const ToastContext = createContext({
+  showToast: false,
+  toastMsg: "",
+  toastClass: "",
+});
+
+const ToastProvider = ({ children }) => {
+  const [toastState, toastDispatch] = useReducer(
+    function toastReducer(toastState, action) {
+      switch (action.type) {
```
Comment (ShekharDhangar) [inline src/context/toast-context.js:17]: Here you can destructure {type,payload} so inside switch case you don't have to access using .

File: src/components/sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+          <NavLink
+            key={item._id}
+            to={item.linkTo}
+            className={({ isActive }) => (isActive ? "active-link" : "")}
+          >
+            <li>
+              <span className="sidebar-icon">{item.icon}</span>
+              {item.linkName}
+            </li>
+          </NavLink>
```
Comment (vrinda-mahajan) [inline src/components/sidebar/Sidebar.jsx:20]: Appreciation:- Good that you are using `Navlink` and `isActive` prop here.

**Review by vrinda-mahajan — APPROVED** [review]
Nothing to suggest. The code is clean and easily understandable. Good to merge.👍
Appreciation:
Good that you have made useAsync custom hook.
Contexts are also well written.
UI is also looking great.

File: src/components/card/VideoCard.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  } = details;
+
+  return (
+    <div className={styles.videoCard}>
+      <div className={styles.cardImageContainer}>
+        <img src={thumbnail} alt="thumbnail" loading="lazy" />
+      </div>
+      <div className={styles.cardBody}>
+        <div className={styles.avatar}>
+          <img src={avatar} alt="avatar" loading="lazy" />
```
Comment (anik31) [inline src/components/card/VideoCard.jsx:22]: appreciation - good that you've added lazy loading

File: src/components/header/Header.jsx [hunk 4]
```diff
@@ -1,38 +1,30 @@
...
+            type="text"
+            placeholder="Search..."
+            className="header-serachbox"
+          />
+        </div>
+        <ul className="nav-links">
+          <li>
+            <a href="#" className="btn btn-primary">
+              Login
+            </a>
```
Comment (anik31) [inline src/components/header/Header.jsx:24]: suggestion - use `Link` instead of `a` tag

File: src/components/sidebar/Sidebar.jsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+export const Sidebar = () => {
+  return (
+    <aside className="sidebar">
+      <ul className="sidebar_links">
+        {sidebarNav.map((item) => (
+          <NavLink
+            key={item._id}
+            to={item.linkTo}
+            className={({ isActive }) => (isActive ? "active-link" : "")}
```
Comment (anik31) [inline src/components/sidebar/Sidebar.jsx:14]: appreciation - good that you've implemented `isActive`

File: src/context/toast-context.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        case ACTION_TYPE_SUCCESS:
+          return {
+            toastClass: "success",
+            toastMsg: action.payload,
+          };
+        case ACTION_TYPE_ERROR:
+          return {
+            toastClass: "error",
+            toastMsg: action.payload,
```
Comment (anik31) [inline src/context/toast-context.js:26]: appreciation - nice implementation of toast

File: src/data/sidebarNav.data.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  WatchLaterOutlineIcon,
+} from "assets/icons/icons";
+import { v4 as uuid } from "uuid";
+
+export const sidebarNav = [
+  {
+    _id: uuid(),
+    linkTo: "/",
+    linkName: "Discover",
+    icon: <HomeIcon />,
```
Comment (anik31) [inline src/data/sidebarNav.data.js:16]: appreciation - liked that you're using a separate file for nav items

File: src/index.js [hunk 8]
```diff
@@ -1,15 +1,22 @@
...
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <VideoProvider>
+      <BrowserRouter>
+        <App />
+      </BrowserRouter>
+    </VideoProvider>
```
Comment (anik31) [inline src/index.js:19]: suggestion - use all providers inside router

**Review by anik31 — COMMENTED** [review]
Good job @VLeads , Code is well structured & UI is also good

**Review by ShekharDhangar — APPROVED** [review]
Nice UI, code is good, Apart from suggested change, rest looks awesome

**Pull request merged**